### PR TITLE
Week1 day2 exercise

### PR DESCRIPTION
**Pull request by yashraj-01**
Finished the exercise for [week1_day2](https://docs.google.com/document/d/1LZ7q3zF9FmQNb2jGD78v5Az5qUtexcFxHWDCTqgtlRA).

### CODE REVIEW

File: week1_day2/exercise/1.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package main
+
+import "fmt"
+
+func calcFreq(s *string, channel chan []int32) {
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go:11]: add comments to describe the function.

File: week1_day2/exercise/1.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	freq := make([]int32, 26)
+	for _, ch := range *s {
+		freq[ch-'a']++
+	}
+	channel <- freq
+}
+
+func main() {
+	arr := []string{"quick", "brown", "fox", "lazy", "dog"}
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go]: remove extra line

File: week1_day2/exercise/1.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	l := len(arr)
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go:22]: remove extra line

File: week1_day2/exercise/1.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	for i := 0; i < l; i++ {
+		temp := <-freqChannel
+
+		for j := 0; j < 26; j++ {
+			res[j] += temp[j]
+		}
+	}
+
+	fmt.Println(res)
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go]: remove extra line

File: week1_day2/exercise/2.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	r := rand.Intn(3)
+	time.Sleep(time.Duration(r) * time.Second)
+	res := rand.Intn(10)
+	fmt.Println(res)
+	return res
+}
+
+func main() {
+	noOfStudents := 200
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/2.go]: remove extra line

File: week1_day2/exercise/2.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	totalRatings := 0
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/2.go:26]: remove extra line

Comment (lokpati-rzp) [reply week1_day2/exercise/2.go:26]: Please add the excercise link here

File: week1_day2/exercise/2.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+		go func() {
+			defer wg.Done()
+			totalRatings += getFeedback()
+		}()
+	}
+
+	wg.Wait()
+
+	avgRating := totalRatings / noOfStudents
+
```
Comment (lokpati-rzp) [inline week1_day2/exercise/2.go]: remove extra line

File: week1_day2/exercise/3.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Transaction struct {
+	id  int32
+	t   int32 // 0: credit, 1: debit
```
Comment (lokpati-rzp) [inline week1_day2/exercise/3.go]: what is "t" , make meaningful name.

File: week1_day2/exercise/3.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+	amt int32
+}
+
+type Account struct {
+	mu      sync.Mutex
+	balance int32
+}
+
+func (a *Account) withdraw(amt int32) {
```
Comment (lokpati-rzp) [inline week1_day2/exercise/3.go]: this function should return golang error. Print the error at call site.

File: week1_day2/exercise/3.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func (a *Account) deposit(amt int32) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	a.balance += amt
+	fmt.Println("Credited:", amt, "Balance:", a.balance)
+}
+
+func (a *Account) transact(t Transaction, wg *sync.WaitGroup) {
```
Comment (lokpati-rzp) [inline week1_day2/exercise/3.go]: rename transact to "txn"

File: week1_day2/exercise/1.go [hunk 11]
```diff
@@ -0,0 +1,38 @@
+package main
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go:1]: please add the task link in each file.

File: week1_day2/exercise/1.go (hunk 1 above)
Comment (yashraj-01) [reply week1_day2/exercise/1.go:11]: Done

File: week1_day2/exercise/1.go (hunk 2 above)
Comment (yashraj-01) [reply week1_day2/exercise/1.go]: Done

File: week1_day2/exercise/1.go (hunk 3 above)
Comment (yashraj-01) [reply week1_day2/exercise/1.go:22]: Done

File: week1_day2/exercise/1.go (hunk 4 above)
Comment (yashraj-01) [reply week1_day2/exercise/1.go]: Done

File: week1_day2/exercise/2.go (hunk 5 above)
Comment (yashraj-01) [reply week1_day2/exercise/2.go]: Done

File: week1_day2/exercise/2.go (hunk 7 above)
Comment (yashraj-01) [reply week1_day2/exercise/2.go]: Done

File: week1_day2/exercise/3.go (hunk 8 above)
Comment (yashraj-01) [reply week1_day2/exercise/3.go]: Renamed to `transactionType`

File: week1_day2/exercise/3.go (hunk 10 above)
Comment (yashraj-01) [reply week1_day2/exercise/3.go]: Done

File: week1_day2/exercise/3.go (hunk 9 above)
Comment (yashraj-01) [reply week1_day2/exercise/3.go]: Done. PTAL.

File: week1_day2/exercise/1.go (hunk 11 above)
Comment (yashraj-01) [reply week1_day2/exercise/1.go:1]: Done

File: week1_day2/exercise/2.go (hunk 6 above)
Comment (yashraj-01) [reply week1_day2/exercise/2.go:26]: Done

File: week1_day2/exercise/1.go [hunk 12]
```diff
@@ -0,0 +1,39 @@
+/*
```
Comment (lokpati-rzp) [inline week1_day2/exercise/1.go]: avoid using adding comments before package name declaration and imports.

Comment (lokpati-rzp) [reply week1_day2/exercise/1.go]: Also do not use /* */ for comments , use "//" for comments.

**Review by lokpati-rzp — APPROVED** [review]
Looks good to me.

**Pull request merged**